### PR TITLE
feat: automatic geag cancellation on unenroll

### PR DESCRIPTION
**Pull request by johnnagro**
### Description

- gated behind a new setting `ENTERPRISE_SUBSIDY_AUTOMATIC_EXTERNAL_CANCELLATION` (for now)
- added tests
- will deploy out to stage and test LIVE ✨

### CODE REVIEW

File: enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py [hunk 1]
```diff
@@ -199,6 +214,16 @@ def handle_reversing_enterprise_course_unenrollment(self, unenrollment):
...
             )
+
+            for external_reference in related_transaction.external_reference.all():
+                provider_slug = external_reference.external_fulfillment_provider.slug
+                if provider_slug == GEAGFulfillmentHandler.EXTERNAL_FULFILLMENT_PROVIDER_SLUG:
+                    # this will raise if there is a problem before we reverse the ledger transaction
+                    # allowing us to try again later
+                    self.geag_client.cancel_enterprise_allocation(external_reference.external_reference_id)
+                else:
+                    raise FulfillmentException(f'dont know how to cancel {provider_slug}')
```
Comment (iloveagent57) [inline enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py:225]: Maybe we should move this into `GEAGFulfillmentHandler`, perhaps adding a new method to the interface like `reverse_fulfillment(transaction)`?

File: enterprise_subsidy/apps/transaction/tests/test_management.py [hunk 2]
```diff
@@ -547,3 +582,310 @@ def test_write_reversals_from_enterprise_unenrollments(
...
+    )
+    @override_settings(ENTERPRISE_SUBSIDY_AUTOMATIC_EXTERNAL_CANCELLATION=False)
+    def test_write_reversals_from_geag_enterprise_unenrollments_disabled_setting(
+        self,
+        mock_signal_client,
+        mock_fetch_course_metadata_client,
+        mock_fetch_recent_unenrollments_client,
+    ):
+        """
+        Test the write_reversals_from_enterprise_unenrollments management command's ability to create a reversal.
```
Comment (iloveagent57) [inline enterprise_subsidy/apps/transaction/tests/test_management.py:605]: nit: this one doesn't actually expect to create a reversal.

File: enterprise_subsidy/apps/transaction/tests/test_management.py [hunk 3]
```diff
@@ -547,3 +582,310 @@ def test_write_reversals_from_enterprise_unenrollments(
...
+    )
+    @override_settings(ENTERPRISE_SUBSIDY_AUTOMATIC_EXTERNAL_CANCELLATION=True)
+    def test_write_reversals_from_geag_enterprise_unenrollments_unknown_provider(
+        self,
+        mock_signal_client,
+        mock_fetch_course_metadata_client,
+        mock_fetch_recent_unenrollments_client,
+    ):
+        """
+        Test the write_reversals_from_enterprise_unenrollments management command's ability to create a reversal.
```
Comment (iloveagent57) [inline enterprise_subsidy/apps/transaction/tests/test_management.py:813]: nit: copypasta

**Review by iloveagent57 — APPROVED** [review]
Small suggestion to make your code ✨ sparkle ✨

File: enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py (hunk 1 above)
Comment (johnnagro) [reply enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py:225]: i really like this and i will implement it, but i am going to incrementally merge this and circle back. i have a few balls in the air and its easier not to let perfection stand in the way of progress.

**Pull request merged**